### PR TITLE
Adding command line options to -ICP to skip translations

### PR DESCRIPTION
**Pull request by HeadLessHUN**
Add suboptions for icp to skip each translation.

-SKIP_TX
-SKIP_TY
-SKIP_TZ

#1898

### CODE REVIEW

File: qCC/ccCommandLineCommands.cpp [hunk 1]
```diff
@@ -6292,23 +6295,27 @@ bool CommandICP::process(ccCommandLineInterface& cmd)
 				QString rotation = cmd.arguments().takeFirst().toUpper();
 				if (rotation == "XYZ")
 				{
-					transformationFilters = CCCoreLib::RegistrationTools::SKIP_NONE;
+					//keep this to not make breaking changes
 				}
 				else if (rotation == "X")
 				{
-					transformationFilters = CCCoreLib::RegistrationTools::SKIP_RYZ;
+					transformationFilters |= CCCoreLib::RegistrationTools::SKIP_RYZ;
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:6302]: why the "|' operator? Did you imagine a use case where -ROT would be used multiple times for a single ICP command? It won't be considered by the registration algorithm anyway... The algorithm only considers one SKIP_R?? option at a time.

File: qCC/ccCommandLineCommands.cpp [hunk 2]
```diff
@@ -6292,23 +6295,27 @@ bool CommandICP::process(ccCommandLineInterface& cmd)
...
+					cmd.print(QObject::tr("[ICP] Skip RYZ, current filter: %1").arg(transformationFilters));
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:6303]: Is it a 'debug' log?

**Review by dgirardeau — CHANGES_REQUESTED** [review]

File: qCC/ccCommandLineCommands.cpp (hunk 2 above)
Comment (HeadLessHUN) [reply qCC/ccCommandLineCommands.cpp:6303]: it was kind of, but it is not that useless :), There were no logs about confirmations of arguments.

File: qCC/ccCommandLineCommands.cpp (hunk 1 above)
Comment (HeadLessHUN) [reply qCC/ccCommandLineCommands.cpp:6302]: you are right, but what if someone send a -SKIP_TX before -ROT?

File: qCC/ccCommandLineCommands.cpp (hunk 2 above)
Comment (dgirardeau) [reply qCC/ccCommandLineCommands.cpp:6303]: Ah ok, it's just that the value of 'transformationFilters' might not be super meaningful to the users... (so maybe the second part could be 'printDebug')

File: qCC/ccCommandLineCommands.cpp (hunk 1 above)
Comment (dgirardeau) [reply qCC/ccCommandLineCommands.cpp:6302]: Ah ok, that's for the SKIP_T?, my bad...

**Comment by dgirardeau** [comment]
Ok, I realized that the 'Debug' versions of the print/warning/error methods of the logger were not readily accessible via the command line parser. So I added them, and moved the logging of 'transformationFilters' in a single final call to printDebug.

I also made sure that explicitly calling '-ROT XYZ' would reset the rotations (just in case).

**Comment by HeadLessHUN** [comment]
> Ok, I realized that the 'Debug' versions of the print/warning/error methods of the logger were not readily accessible via the command line parser. So I added them, and moved the logging of 'transformationFilters' in a single final call to printDebug.
> 
> I also made sure that explicitly calling '-ROT XYZ' would reset the rotations (just in case).

isn't it more resource if we call QObject::tr() without argument or just use the quoted string in this case? (for the future)

**Comment by dgirardeau** [comment]
So 'tr' is to allow translators to translate the string. So it's all a question of whether this is user facing and someone might want to translate it one day. That something that should have done everywhere, but sadly it's missing in a lot of places...

**Comment by HeadLessHUN** [comment]
I think I'm done with this as well.

**Review by dgirardeau — DISMISSED** [review]

**Review by dgirardeau — APPROVED** [review]

**Pull request merged**